Modules/_sqlite/cursor.c

```diff
@@ -1117,14 +1183,20 @@ pysqlite_cursor_iternext(PyObject *op)
         if (self->statement->is_dml) {
             self->rowcount = (long)sqlite3_changes(self->connection->db);
         }
-        (void)stmt_reset(self->statement);
+        int rc = stmt_reset(self->statement);
         Py_CLEAR(self->statement);
+        if (rc != SQLITE_OK) {
+            goto reset_failure;
+        }
     }
     else if (rc != SQLITE_ROW) {
-        set_error_from_db(self->connection->state, self->connection->db);
-        (void)stmt_reset(self->statement);
+        rc = set_error_from_db(self->connection->state, self->connection->db);
+        int reset_ok = stmt_reset(self->statement);
```

Comment:
Nitpick: it's an `int`, not `bool`; would `reset_rc` be better?

Reply:
You mean the "ok" <-> "int"? reset_rc is fine as well.